nixos/modules/security/pam.nix

```diff
@@ -394,6 +394,8 @@ let
               "session optional ${pkgs.gnome3.gnome-keyring}/lib/security/pam_gnome_keyring.so auto_start"}
           ${optionalString (config.virtualisation.lxc.lxcfs.enable)
                "session optional ${pkgs.lxc}/lib/security/pam_cgfs.so -c all"}
+
+          ${optionalString (config.security.pam.defaults != null) ("# Lines appended via security.pam.defaults.\n" + config.security.pam.defaults)}
```

Comment:
I'd use a multiline string here, Placing the entire config into one line is IMHO less readable.

This may look like this then:

``` nix
''
  # ...

  ${optionalString (config.security.pam.defaults != null) ''
    # your extra config
    ${config.security.pam.defaults}
  ''}
''
```

Reply:
Good point, done.